FTP: Add docs for BurstReadFile

hamishwillee opened the pull request:
Adds docs explaining how burstreadfile works. 

Fixes #98

hamishwillee commented on en/services/ftp.md:
```diff
@@ -404,24 +404,57 @@ The GSC should create a timeout after the `RemoveDirectory` command is sent and
+1. Client sends [TerminateSession](#TerminateSession) to close the file once all the chunks have been downloaded
+   The server should send an ACK/NAK, but this may (generally speaking) be ignored by the client.
+
+The client should create a timeout while waiting for a new `BurstReadFile` <!-- and do? -->
```
@bkueng What should the client do if it stops getting BurstReadFile chunks? My assumption is that it has a timeout, and then just starts the normal readfile process, but it could do something else.

bkueng replied:
Yes you can do either one: BurstReadFile or ReadFile

hamishwillee replied:
I understand how this would work for ReadFile - you just send it and wait for ack. But in the BurstReadFile case are you saying that the server just kicks off another whole new session etc? 
Or to put it another way, ReadFile is something you do within a session, but BurstReadFile creates a session - the client only sends it once normally to start a session off - so what would it do in the "restart burst" case caused by an offset?

Does the question make sense?

bkueng replied:
No, BurstReadFile does not create a session. It's very much like ReadFile, just gets you more data.

bkueng replied:
Yes

hamishwillee commented on en/services/ftp.md:
```diff
@@ -404,24 +404,57 @@ The GSC should create a timeout after the `RemoveDirectory` command is sent and
+   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = length of file that has been opened. 
+   - NAK with [error information](#error_codes), e.g. `NoSessionsAvailable`, `FileExists`. 
+     The client may cancel the operation, depending on the error.
+1. Server sends stream of [BurstReadFile](#BurstReadFile) data to client (without ack) until whole file sent, or a server-defined burst size limit is reached.
```
So essentially if you reach this limit then the steaming will stop with burst_complete=0 - the client will be waiting on timeout to proceed.

julianoes commented:
>> the first BurstReadFile sent by the client indicates the offset of the first item in the burst and the total size of the the data it wants to receive in that burst. Is that right?
>
>No, size isn't used in that case.

So what is used? Does it always just read the whole file. For ReadFile (without burst) the size set to the next chunk to send, right?

hamishwillee commented:
>>> the first BurstReadFile sent by the client indicates the offset of the first item in the burst and the total size of the the data it wants to receive in that burst. Is that right?

>>No, size isn't used in that case.

> So what is used? Does it always just read the whole file. For ReadFile (without burst) the size set to the next chunk to send, right?

@julianoes 
The size of the **file** is retrieved for both methods from the OpenFile request, along with the session. Then the client either uses:
- ReadFile to request a _chunk_ of specific sized (up to one message payload) data at an offset. The server responds with ACK message containing said chunk. Process repeated at offsets for whole file.
- BurstReadFile requests a _burst_ of a specific size and offset. Client ACKS just to say "OK" or "error" then streams all the chunks in the range specified by the burst size. Process of requesting bursts repeated until whole file retrieved (missing chunks in bursts can be requested using either a burst or a readfile). 

So they are much the same except you either specify a chunk size or a burst size.

julianoes commented:
Ok thank you @hamishwillee that makes sense.

hamishwillee commented:
Thanks @bkueng 
OK, fixed up and merged. I think good now. 
I also "FYI" updated all the mermaid diagrams to refer to client/server rather than gcs/drones, and added an intro section for reading files that highlights the difference between the methods.
